### PR TITLE
Added message functionality if no data available into the table

### PR DESCRIPTION
**Pull request by devendra940**
1) Fixed code review changes
2) Delete custom table component as we are using only table component 
3) Added functionality show message if not data in table grid  No data available'

### CODE REVIEW

**Comment by devendra940** [comment]
@kaseyllabelle 
1) Fixed reviewed changes
2) Delete custom table component
3) Added functionality No data available

File: src/app/table/table.component.ts [hunk 1]
```diff
@@ -26,10 +26,10 @@ export class TableComponent implements OnInit, OnDestroy {
       { key: 'phone', value: 'Phone' },
       { key: 'description', value: 'Description' }
     ];
-    this.showData();
+    this.showData();    
   }
 
-  showData() {
+   showData() {
```
Comment (kaseyllabelle) [inline src/app/table/table.component.ts]: @devendra940 undo white space change to fix indentation

File: src/app/table/table.component.html [hunk 2]
```diff
@@ -18,12 +18,14 @@
...
     </thead>
-    <tbody>
-        <tr *ngFor="let dataList of allData">
-            <td>{{dataList.date}}</td>
-            <td>{{dataList.amount}}</td>
-            <td>{{dataList.phone}}</td>
-            <td>{{dataList.description}}</td>
-        </tr>
+    <tbody *ngIf="allData.length > 0; else noColumnData">
+            <tr *ngFor="let dataList of allData">
```
Comment (kaseyllabelle) [inline src/app/table/table.component.html]: @devendra940 undo the indentation changes on lines 22-27

File: src/app/table/table.component.html [hunk 3]
```diff
@@ -18,12 +18,14 @@
...
+                <td>{{dataList.date}}</td>
+                <td>{{dataList.amount}}</td>
+                <td>{{dataList.phone}}</td>
+                <td>{{dataList.description}}</td>
+            </tr>
     </tbody>
+    <ng-template #noColumnData>
```
Comment (kaseyllabelle) [inline src/app/table/table.component.html]: @devendra940 the 'no data' message should be in a row nested inside the table body, e.g.:

```
<tbody>
    <tr>
        <td colspan="100%">
            No data available
        </td>
    </tr>
</tbody>
```

File: src/app/table/table.component.scss [hunk 4]
```diff
@@ -82,19 +83,21 @@
         }
     }
 
-    button.th-sortable.asscending {
+    button.th-sortable.ascending
```
Comment (kaseyllabelle) [inline src/app/table/table.component.scss]: @devendra940 please move code for icons into the existing .th-sortable declaration block, lower specificity, and clean up formatting, e.g.:

```
&::after
{
    position: absolute;
    right: 15px;
    font-family: $font-family-icon;
    transform: translateY(2px);
    opacity: .5;
}

&.active
{
    &::after
    {
        opacity: 1;
    }
}

&.ascending
{
    &::after
    {
        content: $icon-caret-up;
    }
}

&.descending
{
    &::after
    {
        content: $icon-caret-down;
    }
}
```
Here's an example file:
[table.component.scss.zip](https://github.com/epsilon-ux/blueprint-table/files/3751065/table.component.scss.zip)

File: src/app/table/table.component.ts [hunk 5]
```diff
@@ -7,7 +7,7 @@ import { TabledataService } from '../service/tabledata.service';
   styleUrls: ['./table.component.scss']
 })
 export class TableComponent implements OnInit, OnDestroy {
-  allData: any;
+  allData: []= [];
```
Comment (kaseyllabelle) [inline src/app/table/table.component.ts]: @devendra940 add 1 white space before the =

File: src/app/table/table.component.ts [hunk 6]
```diff
@@ -26,10 +26,10 @@ export class TableComponent implements OnInit, OnDestroy {
       { key: 'phone', value: 'Phone' },
       { key: 'description', value: 'Description' }
     ];
-    this.showData();
+    this.showData();
```
Comment (kaseyllabelle) [inline src/app/table/table.component.ts]: @devendra940 delete extra white space at the end of this line

File: src/app/table/table.component.ts (hunk 1 above)
Comment (devendra940) [reply src/app/table/table.component.ts]: @kaseyllabelle  This has been fixed

File: src/app/table/table.component.html (hunk 2 above)
Comment (devendra940) [reply src/app/table/table.component.html]: @kaseyllabelle  This has been fixed

File: src/app/table/table.component.html (hunk 3 above)
Comment (devendra940) [reply src/app/table/table.component.html]: @kaseyllabelle  This has been fixed

File: src/app/table/table.component.scss (hunk 4 above)
Comment (devendra940) [reply src/app/table/table.component.scss]: @kaseyllabelle  This has been fixed.
Thanks for this updated code.

File: src/app/table/table.component.ts (hunk 5 above)
Comment (devendra940) [reply src/app/table/table.component.ts]: @kaseyllabelle  This has been fixed

File: src/app/table/table.component.ts (hunk 6 above)
Comment (devendra940) [reply src/app/table/table.component.ts]: @kaseyllabelle  This has been fixed

**Pull request merged**